File: app/src/main/java/com/example/recyclerviewdragdismiss/MainActivity.java
package com.example.recyclerviewdragdismiss;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.WindowManager;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created on 2019/6/14 0014
 * Created by hechao
 * Description
 */
public class MainActivity extends AppCompatActivity {

    private ArrayList<String> mDatas = new ArrayList<>();
    private int height = 0;
    private RecyclerView recyclerView;
    private NewAdapter adapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.recycler_view);

        WindowManager windowManager = (WindowManager) getSystemService(Context.WINDOW_SERVICE);
        int screenWidth = windowManager.getDefaultDisplay().getWidth();
        int screenHeight = windowManager.getDefaultDisplay().getHeight();

        for (int i = 0; i < 10; i++) {
            mDatas.add("Menu " + (i + 1));
        }

        adapter = new NewAdapter(mDatas);
        GridLayoutManager manager = new GridLayoutManager(this, 4);
        recyclerView.setLayoutManager(manager);

        manager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
            @Override
            public int getSpanSize(int i) {
                int viewType = adapter.getItemViewType(i);
                return viewType == 0 ? 1 : 4;
            }
        });
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
    }
}

File: app/src/main/java/com/example/recyclerviewdragdismiss/MainActivity.kt
package com.example.recyclerviewdragdismiss

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created on 2019/6/15 0015
 * Created by hechao
 * Description
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_single.setOnClickListener { startActivity(Intent(this, SingleCategoryActivity::class.java)) }
        btn_multiple.setOnClickListener { startActivity(Intent(this, MultipleCategoryActivity::class.java)) }
    }
}